10674 home mobile view

roseeichelmann opened the pull request:
Fixes cityofaustin/atd-data-tech/issues/10674

This PR branches off of #50 with mobile and tablet design implementations.

**Steps to test:**

[Netlify link](https://deploy-preview-51--atd-product.netlify.app)

1. First test on your normal laptop/desktop viewport.
2. Resize down the screen until the first breakpoint (tablet) where there should be two columns of information for both the We're here to help and Featured sections. Make sure the navbar and footer look good and work properly.
3. Resize down to the smallest breakpoint (mobile) and now everything should be in one column. Make sure the navbar and footer look good and work properly.
4. Extra points if you actually test with your phone and/or tablet

roseeichelmann commented on components/shared/IndexIssueListItem.js:
```diff
@@ -19,33 +20,42 @@ function parseBody(body) {
   const [description, img] = parseBody(issue.body);
+
+  const type = issue.labels.includes("Product Index") ? "product" : "project";
+
```
Moved this here so we don't have to have this code over and over again in `pages/index.js`

roseeichelmann commented on pages/index.js:
```diff
@@ -1,110 +1,67 @@
-          })}
         </Row>
+        <hr className="mt-2 mx-4" />
+        <FeaturedSection />
```
Calling this `FeaturedSection` now instead of `FeaturedCarousel` now that we don't have the carousel in mobile or tablet view

roseeichelmann commented on components/FeaturedSection.js:
```diff
@@ -0,0 +1,90 @@
+          </Row>
+          <Row>
+            {/* This slice is incase there are more than 8 issues with the featured label */}
+            {featuredIssues.slice(0, 8).map((issue) => {
```
We always want to have just 8 otherwise the design is thrown off so this is a safeguard in case an extra featured label is added to an issue

patrickm02L reviewed (commented):
These mobile views are looking so nice!

https://user-images.githubusercontent.com/5566727/199973284-f7c86452-4fe9-44c5-987a-58e0fdf9b3d8.mp4



**Feedback**

:37 - center the "Featured" title in full screen
:52 - dropdown menu works, but needs more space underneath the "get help" button
1:17 - hamburger menu remains activated in open and closed states
1:45 - in the mobile view there looks to be a line break after "tackle" that's rendering on one line
2:32 - really like how when you jump back from an individual project page it goes to the place in the mobile menu where the featured project is
2:48 - inconsistent styling on drop down menu, on full mobile space in between links is expanded

roseeichelmann commented:
@patrickm02L Which do you like better for the hamburger menu, border when selected only or no border at all?

[Screencast from 11-04-2022 10:56:14 AM.webm](https://user-images.githubusercontent.com/67130963/200021044-7e11e129-fa4e-4592-9671-ed90e4664805.webm)

[Screencast from 11-04-2022 10:59:54 AM.webm](https://user-images.githubusercontent.com/67130963/200021303-d2658848-5751-4487-bc0d-ea29454fd1c8.webm)

johnclary commented on components/FeaturedSection.js:
```diff
@@ -0,0 +1,90 @@
+          </Row>
+        </Col>
+      ) : (
+        <Col className="px-0">
```
i'm seeing a horizontal scrollbar when i'm looking at this on desktop / `lg`. i think nixing this `px-0` should clear it up?

<img width="443" alt="Screen Shot 2022-11-04 at 12 56 51 PM" src="https://user-images.githubusercontent.com/14793120/200033011-1eff9c85-0612-4360-bf93-05056a0c979b.png">

roseeichelmann replied:
Huh weird I dont get that, but nixing it bc it doesn't do much!

patrickm02L commented:
> @patrickm02L Which do you like better for the hamburger menu, border when selected only or no border at all?

No border. I think the menu animating will signal that the hamburger menu is an on-off button. 

Also, thanks for fixing the spacing under the get help button.

johnclary reviewed (changes requested):
this is looking really good! i've requested a few minor changes.

i'm also seeing this odd behavior where the "Featured" `<h4>` is rendering left-aligned when i first load the home page on mobile. if i navigate away and then back to the home page, it is center-aligned.

<img width="489" alt="Screen Shot 2022-11-04 at 1 14 19 PM" src="https://user-images.githubusercontent.com/14793120/200036208-0a9a6344-4212-446d-b1d4-01777b4a5718.png">
<img width="494" alt="Screen Shot 2022-11-04 at 1 14 27 PM" src="https://user-images.githubusercontent.com/14793120/200036209-7bf859a5-5422-41f1-ba3a-4c5a732efe76.png">

to be sure, i tested this on my actual iPhone SE and the weirdness happens there as well.

maybe some issue going on with the timing of that `useMediaQuery`?

roseeichelmann commented on components/FeaturedSection.js:
```diff
@@ -0,0 +1,90 @@
+            interval={null}
+            controls={showArrows}
+            onSlide={(e) => setShowArrows(false)}
+            onSlid={(e) => setShowArrows(true)}
```
Good to know!

roseeichelmann commented on components/FeaturedSection.js:
```diff
@@ -0,0 +1,90 @@
+          >
+            <Carousel.Item>
+              <Row className="mb-3">
+                <Col sm={1}></Col>
```
For some reason I thought you didnt need to specify `xs` due to there not being a class infix for it https://react-bootstrap.github.io/layout/breakpoints/ but that doesn't make sense haha, ill change these all to `xs`! I wonder why they render correctly without it

patrickm02L commented on components/FeaturedSection.js:
```diff
@@ -0,0 +1,90 @@
+    <Row>
+      {/* Render rows with featured cards if tablet or mobile else render carousel */}
+      {isTabletOrMobile ? (
+        <Col className="mx-5">
```
Oh yes, I took at the mobile versions and it looks very cramped with padding. Please make the change to no padding like in John's example above.

roseeichelmann commented:
@johnclary I addressed all of your comments, let me know how everything looks! Especially if that weirdness with the "Featured" element is still happening.
@patrickm02L I also removed the padding of the cards so they are not so cramped anymore.

patrickm02L reviewed (approved):
Everything looks phenomenal! It's good to go! 🤘

frankhereford reviewed (approved):
I tested this on my phone, and it looks just right on the money. Thank you @roseeichelmann!
